### PR TITLE
Optimised LMEM memory usage

### PR DESCRIPTION
**Pull request by amit-samal**
Loading batch of genes for individual cell types to memory for lmem analysis.

### CODE REVIEW

**Review by mayur-iipl — APPROVED** [review]

File: scalr/analysis/dge_lmem.py [hunk 1]
```diff
@@ -253,12 +254,14 @@ def plot_lmem_dge_result(self,
...
                     lmem_res_df[pval_col])
-                down_reg_genes_idx = (lmem_res_df[coef_col] < (
-                    self.coef_threshold)) & (lmem_res_df[f'-log10_{pval_col}']
-                                             >= (neg_log10_pval))
-                up_reg_genes_idx = (lmem_res_df[coef_col] > (
-                    self.coef_threshold)) & (lmem_res_df[f'-log10_{pval_col}']
-                                             >= (neg_log10_pval))
+                down_reg_genes_idx = (lmem_res_df[coef_col] <
+                                      (self.coef_threshold)) & (
+                                          lmem_res_df[f'-log10_{pval_col}'] >=
```
Comment (anand-infocusp) [inline scalr/analysis/dge_lmem.py:259]: These changes are introduced due to yapf only, right?

File: scalr/analysis/dge_lmem.py [hunk 2]
```diff
@@ -362,19 +365,19 @@ def generate_analysis(self, test_data: Union[AnnData, AnnCollection],
                 assert cell_type in test_data.obs[
                     self.
                     celltype_column].values, f"{cell_type} must be in the '{self.celltype_column}' column in 'adata.obs'"
+                cell_type_test_data = test_data[test_data.obs[
+                    self.celltype_column] == cell_type]
                 logger.info(f'\nProcessing for "{cell_type}" ...')
                 fixed_val_lmem_result_list = []
-                for batch in range(0, len(test_data.var_names),
+                for batch in range(0, len(cell_type_test_data.var_names),
```
Comment (anand-infocusp) [inline scalr/analysis/dge_lmem.py:372]: Why cell_type? Is it going to be always cell_type for test data?

File: scalr/analysis/dge_lmem.py (hunk 1 above)
Comment (amit-samal) [reply scalr/analysis/dge_lmem.py:259]: Mostly because of yapf,  code changes - 129

**Review by anand-infocusp — APPROVED** [review]
lgtm

**Pull request merged**